### PR TITLE
Task-41630: Make notifications types settings activated by default after upgrade

### PR DESCRIPTION
**Pull request by hassenelkamel-exo**
**- Problem:**  After migration from 6.1.x to 6.3.x, the new notifications types settings of agenda, notes, news, security plugins coming from 6.2.x are not activated by default.
**- Solution:**  We make the old plugin NotificationSettingsUpgradePlugin implemented for agenda and chat notification types settings generic in order to consider any newly implemented notification types (the notifications coming from 6.2.x). The UP should be always executed not only once in order to not miss any newly implemented notification type.

### CODE REVIEW

File: data-upgrade-portal-rdbms/pom.xml [hunk 1]
```diff
@@ -19,9 +19,9 @@
     <version>6.3.x-maintenance-SNAPSHOT</version>
```
Comment (azayati) [inline data-upgrade-portal-rdbms/pom.xml:19]: you should change also the name of the module

File: data-upgrade-portal-rdbms/src/main/resources/conf/portal/configuration.xml [hunk 2]
```diff
@@ -42,18 +42,27 @@
...
           <name>plugin.upgrade.execute.once</name>
           <description>Execute this upgrade plugin only once</description>
-          <value>true</value>
+          <value>false</value>
         </value-param>
         <value-param>
           <name>plugin.upgrade.target.version</name>
           <description>Target version of the plugin</description>
-          <value>6.2.x</value>
+          <value>6.2.0</value>
```
Comment (azayati) [inline data-upgrade-portal-rdbms/src/main/resources/conf/portal/configuration.xml]:
```suggestion
          <value>6.3.0</value>
```
IMHO

File: data-upgrade-portal-rdbms/src/main/resources/conf/portal/configuration.xml [hunk 3]
```diff
@@ -42,18 +42,27 @@
...
         </value-param>
         <value-param>
           <name>plugin.upgrade.async.execution</name>
           <description>Execute this upgrade asynchronously</description>
           <value>true</value>
         </value-param>
+        <value-param>
+          <name>notification.upgrade.settings.plugin</name>
+          <description>Execute this notification plugin Types</description>
+          <value>EventAddedNotificationPlugin, EventModifiedNotificationPlugin, EventCanceledNotificationPlugin,
```
Comment (azayati) [inline data-upgrade-portal-rdbms/src/main/resources/conf/portal/configuration.xml:60]: we should add notes too

File: data-upgrade-portal-rdbms/src/main/java/org/exoplatform/portal/upgrade/notification/NotificationSettingsUpgradePlugin.java [hunk 4]
```diff
@@ -47,12 +43,16 @@ public NotificationSettingsUpgradePlugin(SettingService settingService,
...
+    if (initParams.containsKey(NOTIFICATION_PLUGIN)) {
+      notificationPluginTypes = initParams.getValueParam(NOTIFICATION_PLUGIN).getValue();
+    }
   }
 
   @Override
   public void processUpgrade(String oldVersion, String newVersion) {
     ExoContainer currentContainer = ExoContainerContext.getCurrentContainer();
-
+    List<String>    pluginTypes = Arrays.asList(notificationPluginTypes.replace("\n", "").replaceAll("\\s", "").split(","));
```
Comment (azayati) [inline data-upgrade-portal-rdbms/src/main/java/org/exoplatform/portal/upgrade/notification/NotificationSettingsUpgradePlugin.java]:
```suggestion
    List<String> pluginTypes = Arrays.asList(notificationPluginTypes.replace("\n", "").replaceAll("\\s", "").split(","));
```

File: data-upgrade-portal-rdbms/src/main/resources/conf/portal/configuration.xml [hunk 5]
```diff
@@ -42,18 +42,27 @@
...
-          <value>6.2.x</value>
+          <value>6.2.0</value>
         </value-param>
         <value-param>
           <name>plugin.upgrade.async.execution</name>
           <description>Execute this upgrade asynchronously</description>
           <value>true</value>
         </value-param>
+        <value-param>
+          <name>notification.upgrade.settings.plugin</name>
```
Comment (azayati) [inline data-upgrade-portal-rdbms/src/main/resources/conf/portal/configuration.xml]:
```suggestion
          <name>notification.upgrade.settings.plugin.types</name>
```

**Review by azayati — CHANGES_REQUESTED** [review]
See review

**Comment by ahamdi** [comment]
Could you provide a description for this PR ?

File: data-upgrade-portal-rdbms/pom.xml (hunk 1 above)
Comment (hassenelkamel-exo) [reply data-upgrade-portal-rdbms/pom.xml:19]: it is branch based on feauture/maintenance, i think we shouldn't change it.

**Comment by hassenelkamel-exo** [comment]
> Could you provide a description for this PR ?

done

File: data-upgrade-portal-rdbms/pom.xml (hunk 1 above)
Comment (azayati) [reply data-upgrade-portal-rdbms/pom.xml:19]: why not ?

**Review by azayati — APPROVED** [review]
LGTM

File: data-upgrade-notifications/pom.xml [hunk 6]
```diff
@@ -19,9 +19,9 @@
     <version>6.3.x-maintenance-SNAPSHOT</version>
   </parent>
 
-  <artifactId>data-upgrade-portal-rdbms</artifactId>
+  <artifactId>data-upgrade-plugin-notification</artifactId>
```
Comment (azayati) [inline data-upgrade-notifications/pom.xml:22]: We should rename to data-upgrade-notification

**Review by azayati — COMMENTED** [review]
We should also try to fix the sonar checks code smells

**Pull request merged**